extensions-contrib/druid-jose4j/src/main/java/org/apache/druid/security/jose4j/Jose4jAuthenticator.java

```diff
@@ -0,0 +1,197 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.server.security.AuthConfig;
+import org.apache.druid.server.security.AuthenticationResult;
+import org.apache.druid.server.security.Authenticator;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+
+import javax.annotation.Nullable;
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+
+@JsonTypeName("jose4j")
+public class Jose4jAuthenticator implements Authenticator
+{
+  private static final Logger LOG = new Logger(Jose4jAuthenticator.class);
+
+  private final String name;
+  private final String authorizerName;
+  private final String issuer;
+  private final Set<String> audience;
+  private final String jwksUri;
+  private final boolean skipOnFailure;
+
+  private final JwtAuthenticator jwtAuthenticator;
+
+  @JsonCreator
+  public Jose4jAuthenticator(
+      @JsonProperty("name") String name,
+      @JsonProperty("issuer") String issuer,
+      @JsonProperty("audience") Set<String> audience,
+      @JsonProperty("jwksUri") String jwksUri,
+      @JsonProperty("authorizerName") String authorizerName,
+      @JsonProperty("skipOnFailure") Boolean skipOnFailure
+  )
+  {
+    this.name = Preconditions.checkNotNull(name, "null Name");
+    this.issuer = Preconditions.checkNotNull(issuer, "null issuer");
+    this.jwksUri = Preconditions.checkNotNull(jwksUri, "null jwksUri");
+    this.authorizerName = authorizerName;
+    this.audience = audience;
+    this.skipOnFailure = skipOnFailure == null ? false : skipOnFailure;
+    this.jwtAuthenticator = new JwtAuthenticator(issuer, jwksUri, audience);
+  }
+
+  @Override
+  public Filter getFilter()
+  {
+    return new HTTPAuthenticationFilter();
+  }
+
+  @Override
+  public Class<? extends Filter> getFilterClass()
+  {
+    return HTTPAuthenticationFilter.class;
+  }
+
+  @Override
+  public String getAuthChallengeHeader()
+  {
+    return "Bearer";
+  }
+
+  @Override
+  public Map<String, String> getInitParameters()
+  {
+    return null;
+  }
+
+  @Override
+  public String getPath()
+  {
+    return "/*";
+  }
+
+  @Override
+  public EnumSet<DispatcherType> getDispatcherType()
+  {
+    return null;
+  }
+
+
+  @Override
+  @Nullable
+  public AuthenticationResult authenticateJDBCContext(Map<String, Object> context)
+  {
+    String bearerToken = (String) context.get("token");
```

Comment:
is this consistent with what other extensions are doing or should we use something more explicit such as bearer-token or namespace it based on auth extension to avoid clashing with existing context fields?

Reply:
From [SQL JDBC driver API](https://druid.apache.org/docs/latest/querying/sql-jdbc.html), 
> We recommend using Avatica JDBC driver version 1.17.0 or later. Note that as of the time of this writing, Avatica 1.17.0, the latest version, does not support passing connection string parameters from the URL to Druid, so you must pass them using a Properties object.

so extensions parse pre-defined keys from JDBC connector context. I see people using different key names for bearer tokens online.  But I agree with you, we need a name to avoid clashing. We should have a README page and explicitly state 'bearer-token' context we parse